gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -124,18 +126,27 @@ private static RestHighLevelClient buildElasticSearchClient(Parameters parameter
                             "http")));
   }
 
-  private static void index(RestHighLevelClient client, Path file, String indexName) throws IOException {
+  private static void index(RestHighLevelClient client, Path file, String indexName, String corpusFormat) throws IOException {
     log.info("Indexing {}", file.toAbsolutePath());
 
-    // If file ends with .xml.gz, this may be the wrong version of Gigaword.
-    if (file.toString().toLowerCase().endsWith(".xml.gz")) {
-      log.warn("Indexing file ending with .xml.gz. This may indicate incorrect Gigaword version. "
-          + "Make sure you are using the plain version LDC2011T07.");
+    Iterable<Article> concatenatedGigawordDocuments = null;
+    if (corpusFormat.equals("annotated_gigaword")) {
+      log.warn("Indexing an annotated version of Gigaword. Please note that this procedure does not preserve " +
+              "the offsets from the original document texts and therefore cannot be used for " +
+              "the curated training process.");
+      concatenatedGigawordDocuments = ConcatenatedAnnotatedGigawordDocuments.fromAnnotatedGigwordGZippedFile(file);
+    }
+    else if (corpusFormat.equals("gigaword")) {
+      concatenatedGigawordDocuments = ConcatenatedGigawordDocuments.fromGigwordGZippedFile(file);
+    }
+    else {
+      System.err.println(USAGE);
```

Comment:
Can you put an explicit error here that the `corpusFormat` is unknown? Otherwise the user will not be sure what's wrong.

Reply:
Changed in latest commit.